File: 0830.positions_of_large_groups/solution.py
from typing import List


class Solution:
    def large_group_positions(self, s: str) -> List[List[int]]:
        ans, left, right = [], 0, 0
        for i in range(len(s)):
            if s[i] == s[right]:
                right = i
            else:
                left = right = i
            if (
                right - left > 1 and s[right] != s[right + 1 : right + 2]
            ):  # tip: ignore out of index error
                ans.append([left, right])
        return ans


if __name__ == "__main__":
    assert Solution().large_group_positions("abbxxxxzzy") == [[3, 6]]
    assert Solution().large_group_positions("aaa") == [[0, 2]]

File: 0830.positions_of_large_groups/solution.go
package leetcode0830

func largeGroupPositions(S string) [][]int {
	var ans [][]int
	var i int
	for j := 0; j < len(S); j++ {
		if j == len(S)-1 || S[j] != S[j+1] {
			if j-i > 1 {
				ans = append(ans, []int{i, j})
			}
			i = j + 1
		}
	}
	return ans
}
